### ISSUE TITLE
No items is show after clearing the search box

### ISSUE THREAD

**Issue by gagarine** [MacPass/MacPass#942]
When searching, removing the content of the search field will show no items.

**To Reproduce**
1. Write some stuff in the search box
2. Erase those stuff
3. No item is shown

**Expected behavior**
Has the search box is empty, I except to see all the items

**Desktop (please complete the following information):**
 - macOS: 10.14.5
 - Version 0.7.9

**Additional context**
I believe we should not have a two listing view (search and "normal") but only one, simply display all the item when the search box is empty. I noticed a lot of strange behaviors in "search mode" like when i try to add an item by drag&drop from the browser. Also suddenly showing the "group" is kind of strange as I do not use this features.

**Comment by mstarke** [comment]
Reworking how search is handled is something that might be done combined with some more interface changes. This also could add search for groups in the outline view and much more other things.

About the "Search mode" this seemed to be the reasonable solutions (and I was trying to mirror the behaviour of the Finder).

Regarding an empty but active search I would consider the current setup correct, otherwise a simple "display everything if search is empty" might work as well.

**Comment by georgesnow** [comment]
I would definitely prefer the “display everything if search is empty” 😬

**Comment by gagarine** [comment]
The things is the Finder is a search (you execute a query and it return the result) but in MacPass it should be a filter (you have list of things and you start filtering it).

> About the "Search mode" this seemed to be the reasonable solutions (and I was trying to mirror the behaviour of the Finder).
If you look at the Keychain or Console,  they do not have two modes (as they are filter and not search).

> Regarding an empty but active search I would consider the current setup correct, otherwise a simple "display everything if search is empty" might work as well.
Again, it's because it's a filter that when empty it should show everything.

**Comment by mstarke** [comment]
@gagarine the search in finder is a "filter on all your files" so I guess one could argue both ways. And the UI clearly marks this as "Search" not "Filter list" so I would again argue that the search should only yield results if there are any (that is, matches) but I guess this argument can be used both ways.

**Comment by mstarke** [comment]
@gagarine this is not the intended behaviour. Seems like a regression.

**Comment by gagarine** [comment]
It should not be fixed by https://github.com/buddax2/MacPass/commit/5a4eb302280410c7dd52e9f5bd5b09bc94c51db5 ?

**Comment by mstarke** [comment]
> Now it's even worst (Version 0.7.10 (22740)). You can clear close search mode.

To clarify, this is no regression. The behaviour is exactly the same in 0.7.9

**Comment by gagarine** [comment]
Also, MacPass keep the active entry in the right panel active even if it's not listed in the search result. Again the Contact app has better behaviour in this case also by switching to the first result

**Comment by georgesnow** [comment]
@mstarke you are right in Mail.app the behavior matches 0.7.10. when you press ESC it never lets you leave the search box.  so what's the best way....🤷‍♂️

**Comment by gagarine** [comment]
> you search something, change your mind, delete everything and then want to type a new search, you have to re-enter the search again.

Ok this is the misunderstanding we have. I'm not saying the cursor should go away of the field! Just the search mode (the view of the entry) should just display all entries when the search is empty.

**Comment by gagarine** [comment]
You can quickly end up in this situation (add some keyword, then delete them). I don't see how this can be user-friendly.
![Screenshot 2020-02-13 at 01 22 54](https://user-images.githubusercontent.com/164074/74386767-80230780-4dff-11ea-86cf-14822d4b178e.png)

I'm 200% confident that the correct way is to similar pattern than the Contacts.app. Is there any downside? I can't see one.

**Why this Search Fields is not a search?**
Because search function are application than go outside of their context. Take the finder -> you do not filter element in the windows but search either you entire mac or all the subfolder of the current folder. => no input = no search

In the case of MacPass, we have a list of item that we want to filter.

It's the difference between "grep" (The grep utility searches any given input files) and "find" ( The find utility recursively descends the directory tree for each path listed).

The only question would be related to groups. I have no group. 
![Screenshot 2020-02-13 at 01 40 33](https://user-images.githubusercontent.com/164074/74387695-e577f800-4e01-11ea-95df-542c86743d59.png)

So do you want to filter only the selected group? I guess so. Why using groups otherwise?

**Comment by vitopepito** [comment]
I just wanted to add to this relatively old issue. On the release build 0.7.12 the search works quite well. However, on the continuous build, pressing 'esc' with the search box in focus gives an odd response, similar to what was discussed in this thread. The search box is unfocused but no results are shown and an 'error sound' is played.
